scripts/vehicle_node.py

```diff
@@ -150,54 +149,61 @@ def update_position(self, event):
                 innerlist.append(self.gaussian_array[i][j])
                 cloud.append(innerlist)
 
-        #Create interpolation function based on the ranges and gaussian data
+        #3. Create interpolation function based on the ranges and gaussian data
         f = RectBivariateSpline(x,y, self.gaussian_array)
 
-        #Create the footprint for the vehicle
-        #TODO implement footprint that changed with yaw of vehicle
-        x1 = np.linspace((self.pose.position.x-self.vehicle_length), 
-                        (self.pose.position.x+self.vehicle_length),15)
-        y1 = np.linspace((self.pose.position.y-self.vehicle_width), 
-                        (self.pose.position.y+self.vehicle_width),15)
-
-        x1, y1 = np.meshgrid(x1, y1)
-        x1 = x1.ravel()
-        y1 = y1.ravel()
-
-        #Interpolate z values for points in the footprint of the vehicle
-        z_points = f(x1, y1, grid = False)
-
-        #Calculate the average of the points in the footprint and assign it to the z position of the vehicle.
-
-        avg = np.sum(z_points)/len(z_points)
-        
-        self.pose.position.z = avg
+        #4. Find z value for x and y coordinate of vehicle using interpolation function
+        self.pose.position.z = f(self.pose.position.x, self.pose.position.y)
 
         #Convert Euler Angles to Quarternion
         q = tf.transformations.quaternion_from_euler(0.0, 0.0, self.vehicle_yaw)
 
         #Broadcast vehicle frame which is a child of the world frame
-
-        self.br.sendTransform((self.pose.position.x, self.pose.position.y, self.pose.position.z), 
+        br = tf.TransformBroadcaster()
+        br.sendTransform((self.pose.position.x, self.pose.position.y, self.pose.position.z), 
                         q, rospy.Time.now(),"vehicle_frame", "map")
 
-        point = [self.pose.position.x, self.pose.position.y, self.pose.position.z]
-        trans1_mat = tf.transformations.translation_matrix(point)
-        rot1_mat   = tf.transformations.quaternion_matrix(q)
-        mat1 = np.dot(trans1_mat, rot1_mat)
+        #Construct the homogenous transformation matrix for map to vehicle frame
+        translation = [self.pose.position.x, self.pose.position.y, self.pose.position.z]
+        map_t_vehicle = tf.transformations.translation_matrix(translation)
```

Comment:
Looking at this now I realise that you already have the translation and since `map_R_vehicle` is a transformation matrix with only the rotation we can just put the translation into that matrix instead of doing these slicings and unnecessary conversions.  

`map_T_vehicle   = tf.transformations.quaternion_matrix(q)
 `
`map_T_vehicle[:3,3] = np.array(translation)
`